api/provider/aws/test_aws/deploy_create_test.go

```diff
@@ -0,0 +1,128 @@
+package test_aws
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/golang/mock/gomock"
+	provider "github.com/quintilesims/layer0/api/provider/aws"
+	"github.com/quintilesims/layer0/api/tag"
+	awsc "github.com/quintilesims/layer0/common/aws"
+	"github.com/quintilesims/layer0/common/config/mock_config"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDeployCreate(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+	defer provider.SetEntityIDGenerator("dpl_id")()
+
+	tags := models.Tags{
+		{
+			EntityID:   "dpl_id",
+			EntityType: "deploy",
+			Key:        "name",
+			Value:      "dpl_name",
+		},
+		{
+			EntityID:   "dpl_id",
+			EntityType: "deploy",
+			Key:        "version",
+			Value:      "dpl_version",
+		},
+		{
+			EntityID:   "dpl_id",
+			EntityType: "deploy",
+			Key:        "arn",
+			Value:      "arn:aws:ecs:region:012345678910:task/arn",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// define container defaults
+	container := &ecs.ContainerDefinition{}
+	container.SetName("test_name")
+	logConfig := &ecs.LogConfiguration{
```

Comment:
we should probably have 2 containers, so we verify that one w/o log configuration will add the default layer0 log configuration.

Reply:
At this point, I can't figure out a decent way to make this test capture that case -- in `deploy_create.go`, the first step a deploy goes through is to have the request rendered via unmarshaling into an `ecs.TaskDefinition`. From there, we check if the subfield `LogConfiguration` is nil, and if so, we inject the proper config. 

Using the mocks, on the other hand, forces us to use `SetLogConfiguration` when creating an `ecs.ContainerDefinition` or you get a `no matching expected call: *mock_config.MockAPIConfig.LogGroupName([])` error. Setting `LogConfiguration` to `nil` doesn't make a difference. If you have a good idea on how to get around that, LMK